Showcase multilingual dense retrieval with E5

joshdevins opened the pull request:
This new notebook shows a simple walkthrough of using the E5 multilingual embedding model to query a mixed language corpus. It follows the same level of abstraction and details as the other notebooks in this directory, opting to perform all inference outside of the stack. We use select passages from the MIRACL dataset in English and German.

leemthompo commented:
👋 @joshdevins as part of https://github.com/elastic/elasticsearch-labs/pull/8 we renamed the notebooks folder to simply `notebooks`, if you don't mind targeting that folder!

joemcelroy commented on colab-notebooks-examples/search/04-multilingual.ipynb:
```diff
@@ -0,0 +1,487 @@
+   "metadata": {
+    "id": "Y01AXpELkygt"
+   },
+   "source": [
```
we can drop these requirements here. We have done for most recent notebook updates + only calling out ML node needs for when model is used in stack

joemcelroy commented on colab-notebooks-examples/search/04-multilingual.ipynb:
```diff
@@ -0,0 +1,487 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pretty_response(query(\"health\"))"
```
suggestion: include the std_out for these responses.

joemcelroy commented on colab-notebooks-examples/search/04-multilingual.ipynb:
```diff
@@ -0,0 +1,487 @@
+   ]
+  },
+  {
+   "cell_type": "code",
```
typo: empty code frame.

leemthompo commented on colab-notebooks-examples/search/04-multilingual.ipynb:
```diff
@@ -0,0 +1,487 @@
+    "id": "gEzq2Z1wBs3M"
+   },
+   "source": [
+    "[TODO: Update]\n",
```
```suggestion
    "\n",
```

joshdevins replied:
I copied that from another notebook. We might need to clean this up elsewhere as well!

leemthompo commented on colab-notebooks-examples/search/04-multilingual.ipynb:
```diff
@@ -0,0 +1,487 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that in the results above, we see that the document about healthcare,\n",
```
Might need another German speaker to verify the results 😄

joshdevins replied:
I am a German speaker 😉

joshdevins commented:
@joemcelroy @leemthompo I've rebased and should have all the new stuff from last week. I've also addressed all of the comments from the older version. I made sure to update READMEs and tried to normalise as much as I could see between all the notebooks in `search`. Please have another pass and let me know if this looks good! I'm hoping to complete a blog post this week before PTO and I'll be linking to this notebook.

leemthompo commented on notebooks/search/01-keyword-querying-filtering.ipynb:
```diff
@@ -1,14 +1,5 @@
 {
   "cells": [
-    {
-      "cell_type": "markdown",
-      "metadata": {
-        "id": "0wgbLWl2udLQ"
-      },
-      "source": [
-        "[Quick Start](https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/search/00-quick-start.ipynb) || **Keyword Querying Filtering** || [Hybrid search with RRF](https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/search/02-hybrid-search.ipynb) || [ELSER](https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/search/03-ELSER.ipynb)"
-      ]
-    },
```
@joshdevins is just a drive-by cleanup? :)

joshdevins replied:
Yeah, I wanted to normalise all the notebooks and this one stood out as having a header that none of the others had.

leemthompo reviewed (approved):
LGTM, consider adding a blurb in the `search` notebooks [README](https://github.com/elastic/elasticsearch-labs/tree/main/notebooks/search), probably just a condensed bulleted version of your blog intro will do the trick. 

Just started adding those category readmes incrementally this week. :)
